### DISCUSSION TITLE
Order of arguments in read_resource in frictionless-r

### DISCUSSION THREAD

**Discussion by fjuniorr** [frictionlessdata/frictionlessdata.io#705] [Ideas] [answered]
I've noted that the order of the arguments have changed in `read_resource` from `(package, resource_name)` to `(resource_name, package)` in https://github.com/frictionlessdata/frictionless-r/commit/ba943b13e123d0b05575fb036054f5ac53e4c17a.

One of the advantages of the previous ordering is that we could write code with both base (`|>`) and magrittr (`%>%`) pipes in the following style:

```R
package |> read_resource("resource-name")
```

With the new order, with magrittr pipes we can still use `.` as placeholder

```R
# still not as pleasant
package %>% read_resource("resource-name", .) 
```

but with the base pipe operator I need

```R
# not fun
package |> (function(x) read_resource("resource-name", x))() 
```

I couldn't find the reasoning behind the change, but wanted to chime in in the discussion for the design of [`frictionless-r`](https://github.com/frictionlessdata/frictionless-r).

**Comment by peterdesmet** [comment, marked as answer]
Excellent question, I've just opened discussions at frictionless-r. Can you re-post it at https://github.com/frictionlessdata/frictionless-r/discussions and close it here?

**Reply by fjuniorr** [reply]
Done!

Closed in favor of https://github.com/frictionlessdata/frictionless-r/discussions/52
